### PR TITLE
lib: modem_fota: changes for nRF Cloud prod

### PR DESCRIPTION
**Pull request by jayteemo**
Use static ip addresses and prod JITP endpoint address.

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

**Comment by jayteemo** [comment]
@tokangas please confirm intended behavior when hostname is overridden.

File: lib/modem_fota/fota_client_mgmt.c [hunk 1]
```diff
@@ -68,25 +69,27 @@ static void http_response_cb(struct http_response *rsp,
...
 static int do_connect(int * const fd, const char * const hostname,
-		      const uint16_t port_num, bool use_fota_apn);
+		      const uint16_t port_num, bool use_fota_apn,
+		      const char * const ip_address);
 static int parse_pending_job_response(const char * const resp_buff,
 				      struct fota_client_mgmt_job * const job);
 
-#define API_HOSTNAME "static.api.beta.nrfcloud.com"
+#define API_STATIC_IP_1 "75.2.37.83"
+#define API_STATIC_IP_2 "99.83.231.82"
```
Comment (junqingzou) [inline lib/modem_fota/fota_client_mgmt.c:78]: Should these two IP addresses be set by CONFIG? Recommended to use host name **#if !defined CONFIG_MODEM_FOTA_CARRIER_LIB**  #else set the host by IP address that is defined with a CONFIG item

**Comment by tokangas** [comment]
> @tokangas please confirm intended behavior when hostname is overridden.

Hostname override has so far been used for testing cases where the server is unreachable. So in principle it would be enough that the TCP connection setup uses address given with hostname override, but it would be good that also the HTTP request and SNI use the given hostname.

File: lib/modem_fota/fota_client_mgmt.c (hunk 1 above)
Comment (tokangas) [reply lib/modem_fota/fota_client_mgmt.c:78]: I think we could have a normal Kconfig item for static IP usage. I would like to have as little carrier lib specific stuff in the library as possible.

Do we need to make the IP addresses configurable? The idea is, after all, that they're static.

I will merge the changes now, we can change the configuration separately if needed.

Comment (junqingzou) [reply lib/modem_fota/fota_client_mgmt.c:78]: The idea is to not expose the IP addresses on GitHub open source

Comment (jayteemo) [reply lib/modem_fota/fota_client_mgmt.c:78]: the IP addresses are not secret, anyone can execute `nslookup static.api.nrfcloud.com` and see them.

**Pull request merged**